server/item/crossbow.go

```diff
@@ -0,0 +1,176 @@
+package item
+
+import (
+	"github.com/df-mc/dragonfly/server/block/cube"
+	"github.com/df-mc/dragonfly/server/item/potion"
+	"github.com/df-mc/dragonfly/server/world"
+	"time"
+	_ "unsafe"
+)
+
+// Crossbow is a ranged weapon similar to a bow that uses arrows or fireworks as ammunition.
+type Crossbow struct {
+	// Item is the item the crossbow is charged with.
+	Item Stack
+}
+
+// Charge starts the charging process and prints the intended duration.
+func (c Crossbow) Charge(releaser Releaser, tx *world.Tx, ctx *UseContext, duration time.Duration) {
+	if !c.Item.Empty() {
+		return
+	}
+
+	creative := releaser.GameMode().CreativeInventory()
+	held, left := releaser.HeldItems()
+
+	chargeDuration := time.Duration(1.25 * float64(time.Second))
+	for _, enchant := range held.Enchantments() {
+		if q, ok := enchant.Type().(interface{ ChargeDuration(int) time.Duration }); ok {
+			chargeDuration = q.ChargeDuration(enchant.Level())
```

Comment:
This should probably be `chargeDuration = min(chargeDuration, q.ChargeDuration(enchant.Level())` instead of the charge duration being overwritten by some arbitrary enchant ordering